envi/__init__.py

```diff
@@ -1351,71 +1466,112 @@ def getCurrentArch():
 
     return ret
 
+def getRealArchName(name):
+    for arch, adict in arch_defs.items():
+        rname = adict.get('name')
+        if name == rname:
+            return rname
+
+        if name in adict.get('aliases', []):
+            return rname
+
+    return name
+
+def getArchNames():
+    """
+    Return a dict of arch_names which are currently implemented
+    """
+    modules = getArchModules(new=False)
+    return {arch: name for (name, arch) in arch_by_name.items() if not arch_defs.get(arch).get('disabled')}
+    
+
 def getArchModule(name=None):
     """
-    return an Envi architecture module instance for the following
+    Return an Envi architecture module instance for the following
     architecture name.
 
-    Current architectures include:
-
-    i386 - Intel i386
-    amd64 - The new 64bit AMD spec.
+    If name is None, uses the "current" architecture.
     """
     if name is None:
         name = getCurrentArch()
 
-    # Some builds have x86 (py2.6) and some have other stuff...
-    if name in ['i386', 'i486', 'i586', 'i686', 'x86']:
-        import envi.archs.i386 as e_i386
-        return e_i386.i386Module()
+    rname = getRealArchName(name)
+    archnum = arch_by_name.get(rname)
 
-    elif name in ('amd64', 'x86_64'):
-        import envi.archs.amd64 as e_amd64
-        return e_amd64.Amd64Module()
+    if archnum not in arch_defs or arch_defs.get(archnum).get('disabled'):
+        raise ArchNotImplemented(name, "Arch not defined, or Arch Disabled")
```

Comment:
We can be more specific about whether it's disabled or not defined here.

Reply:
done